pkg/plugin/plugin_test.go

```diff
@@ -0,0 +1,142 @@
+	return t.rPluginGithubIssue, t.rErr
+}
+
+func TestValidate(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name      string
+		validator *testIssueMatcher
+		req       *jvspb.ValidateJustificationRequest
+		wantResq  *jvspb.ValidateJustificationResponse
+		wantErr   string
+	}{
+		{
+			name: "success",
+			validator: &testIssueMatcher{
+				rPluginGithubIssue: &pluginGitHubIssue{
+					Owner:       "test-owner",
+					RepoName:    "test-repo-name",
+					IssueNumber: 1,
+				},
+				rErr: nil,
+			},
+			req: &jvspb.ValidateJustificationRequest{
+				Justification: &jvspb.Justification{
+					Category: "github",
+					Value:    "https://github.com/test-owner/test-repo/issues/1",
+				},
+			},
+			wantResq: &jvspb.ValidateJustificationResponse{
+				Valid: true,
+				Annotation: map[string]string{
+					respAnnotationKeyIssueOwner:  "test-owner",
+					respAnnotationKeyIssueRepo:   "test-repo-name",
+					respAnnotationKeyIssueNumber: "1",
+				},
+			},
+		},
+		{
+			name: "internal_error",
+			validator: &testIssueMatcher{
+				rPluginGithubIssue: nil,
+				rErr:               fmt.Errorf("injected error"),
+			},
+			req: &jvspb.ValidateJustificationRequest{
+				Justification: &jvspb.Justification{
+					Category: "github",
+					Value:    "https://github.com/test-owner/test-repo/issues/1",
+				},
+			},
+			wantErr: "injected error",
+		},
+		{
+			name: "wrong_category",
+			validator: &testIssueMatcher{
+				rPluginGithubIssue: nil,
+				rErr:               nil,
+			},
+			req: &jvspb.ValidateJustificationRequest{
+				Justification: &jvspb.Justification{
+					Category: "test-category",
+					Value:    "https://github.com/test-owner/test-repo/issues/1",
+				},
+			},
+			wantResq: &jvspb.ValidateJustificationResponse{
+				Valid: false,
+				Error: []string{`failed to perform validation, expected category "test-category" to be "github"`},
+			},
+		},
+		{
+			name: "issue_not_found",
+			validator: &testIssueMatcher{
+				rPluginGithubIssue: nil,
+				rErr:               errors.Join(errInvalidJustification, fmt.Errorf("issue not found")),
+			},
+			req: &jvspb.ValidateJustificationRequest{
+				Justification: &jvspb.Justification{
+					Category: "github",
+					Value:    "https://github.com/test-owner/test-repo/issues/1",
+				},
+			},
+			wantResq: &jvspb.ValidateJustificationResponse{
+				Valid: false,
+				Error: []string{"invalid justification\nissue not found"},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		ctx := context.Background()
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			p := &GitHubPlugin{
+				validator: tc.validator,
+			}
+			gotResq, gotErr := p.Validate(ctx, tc.req)
+			if diff := testutil.DiffErrString(gotErr, tc.wantErr); diff != "" {
+				t.Errorf(diff)
+			}
+			if diff := cmp.Diff(tc.wantResq, gotResq, cmpopts.IgnoreUnexported(jvspb.ValidateJustificationResponse{})); diff != "" {
```

Comment:
with cmpopts.IgnoreUnexported(jvspb.ValidateJustificationResponse{}), does it compare ValidateJustificationResponse?

Reply:
Yes, it only ignore the unexported field in `jvspb.ValidateJustificationResponse`, the rest of the fields will still be compared.

Reply:
I see, thanks!